### PR TITLE
feat: SelectDocfields as GQL Enums

### LINKED ISSUE #23: Generating GraphQL enums on a Frappe Select Field
I believe it should possible to generate GraphQL types on a field like select, but do mention your thoughts on this.

Here for example, I rely on address type, which only has these values as `option`. So, Ideally we can generate type for this like so; `AddressType.Billing` etc..

|  image |  image |
|---|---|

### PR DESCRIPTION
**Pull request by fahimalizain**
- Updated generate_sdl to make enum types for select docfield options
- Updated document resolver to return select values scrubbed and capitalized
- Updated DocType.graphql since that was the only core doctype that had Select fields in them.

Fixes #23

### CODE REVIEW

File: frappe_graphql/utils/generate_sdl/doctype.py [hunk 1]
```diff
@@ -160,6 +180,12 @@ def get_graphql_type(docfield):
         graphql_type = "BaseDocType"
     elif docfield.fieldtype in table_fields:
         graphql_type = f"[{docfield.options.replace(' ', '')}!]!"
+    elif docfield.fieldtype == "Select":
+        graphql_type = get_select_docfield_enum_name(meta.name, docfield)
+
+        # Mark NonNull if there is no empty option
+        if all([len(x or "") > 0 for x in (docfield.options or "").split("\n")]):
+            graphql_type += "!"
```
Comment (e-lobo) [inline frappe_graphql/utils/generate_sdl/doctype.py:214]: hmmm i think only check meta if the select `docfield` **is_mandatory** and append the `!`

Comment (fahimalizain) [reply frappe_graphql/utils/generate_sdl/doctype.py:214]: resolved 👍

**Review by e-lobo — APPROVED** [review]

**Comment by fahimalizain** [comment]
Guys- since the PR is so old, had to rebase this to master.

File: frappe_graphql/utils/generate_sdl/__init__.py [hunk 2]
```diff
@@ -24,27 +24,33 @@
 
 
 def make_doctype_sdl_files(target_dir, app=None, modules=[], doctypes=[],
-                           ignore_custom_fields=False):
+                           ignore_custom_fields=False, disable_enum_selectdf=False):
```
Comment (assemmarwan) [inline frappe_graphql/utils/generate_sdl/__init__.py]: I think better to rename `disable_enum_selectdf` => `disable_enum_select_fields` to be more readable and consistent with the option in the CLI

Comment (fahimalizain) [reply frappe_graphql/utils/generate_sdl/__init__.py]: @assemmarwan fixed 👍🏼

**Review by assemmarwan — APPROVED** [review]

**Pull request merged**